tests/conftest.py

```diff
@@ -250,26 +250,47 @@ def kafka_topic_factory():
     from kafka import KafkaAdminClient
     from kafka.admin import NewTopic
     from kafka.errors import UnknownTopicOrPartitionError
+    import logging
+
+    logger = logging.getLogger(__name__)
 
     load_dotenv()
 
     bootstrap_servers = getenv(
         "KAFKA_BOOTSTRAP_SERVERS", type=list, separator=","
     )
+    if bootstrap_servers:
+        security_protocol = getenv("KAFKA_SECURITY_PROTOCOL")
+        if security_protocol == "SSL":
+            args = {
+                "security_protocol": security_protocol,
+                "ssl_cafile": getenv("KAFKA_SSL_CAFILE"),
+                "ssl_certfile": getenv("KAFKA_SSL_CERTFILE"),
+                "ssl_keyfile": getenv("KAFKA_SSL_KEYFILE"),
+            }
+        else:
+            args = {}
+
+        admin = KafkaAdminClient(bootstrap_servers=bootstrap_servers, **args)
+
+        @contextmanager
+        def create_kafka_topic(topic_name):
+            try:
+                admin.delete_topics([topic_name])
+            except UnknownTopicOrPartitionError:
+                pass
+
+            admin.create_topics([NewTopic(topic_name, 1, 1)], timeout_ms=1500)
+            yield
 
-    admin = KafkaAdminClient(bootstrap_servers=bootstrap_servers)
-
-    @contextmanager
-    def create_kafka_topic(topic_name):
-        try:
             admin.delete_topics([topic_name])
-        except UnknownTopicOrPartitionError:
-            pass
+            admin.close()
 
-        admin.create_topics([NewTopic(topic_name, 1, 1)], timeout_ms=1500)
-        yield
+        return create_kafka_topic
 
-        admin.delete_topics([topic_name])
-        admin.close()
+    else:
+        logger.warning(
+            f"There is no KAFKA_BOOTSTRAP_SERVERS. "
+            f"Creation of kafka_topic passed."
```

Comment:
you probably want to use "skipped" instead of "passed" here

Reply:
Corrected.